Sources/MapboxNavigation/NavigationOptions.swift

```diff
@@ -47,6 +47,12 @@ open class NavigationOptions: NavigationCustomizable {
      */
     open var predictiveCacheOptions: PredictiveCacheOptions?
```

Comment:
Good catch! Even though `NavigationMapView.enablePredictiveCaching` is covered by constructor, `PredictiveCacheManager` is publicly available and thus can potentially be called before any other setup is done.

Reply:
It worth thinking about how to avoid such potential problems in the future. Maybe it can be solved by extracting Navigator pre-initialization config into a separate struct to make a single entry point that at least easier to validate in Xcode by inspecting "show all callers" or something like that.

Reply:
I thought about that, but I see there are few issues: either we force this struct to be configured before any interaction w/ SDK (technically we introduce an SDK setup step) and trust user to do so at proper moment, or we encapsulate this setup anywhere where it may be triggered which makes the process transparent for users, but requires us to maintain all such places. What we are doing currently looks more like a latter case.

Reply:
I understand why you've done it this way. What I'm talking about is to make future internal changes safer, to reduce the chance of forgetting to put property initiation into some of the places where a Navigator instance can be created.

Reply:
I think we should better frame it outside current PR, because this seems to me like a complex solution, covering more than tile store configuration.